src/imgtools/io/sample_output.py

```
@@ -131,43 +181,52 @@
-                logger.error(errmsg)
-                raise TypeError(errmsg)
-        return saved_files
+                else:
+                    errmsg = (
+                        f"Unsupported image type: {type(image)}. "
+                        "Expected MedImage or VectorMask."
+                    )
+                    logger.error(errmsg)
+                    raise TypeError(errmsg)
+            except Exception as e:
+                errmsg = f"Failed to save image SeriesUID: {image.metadata.get('SeriesInstanceUID', 'unknown')}: {e}"
```

Pylint (3.3.7)

[convention] Line too long (117/100)

(C0301)